apps/spruce/src/pages/version/VersionTabs/index.tsx

```diff
@@ -184,67 +182,62 @@ export const VersionTabs: React.FC<VersionTabProps> = ({ version }) => {
       numFailedChildPatches,
       numStartedChildPatches,
       numSuccessChildPatches,
-      versionId: versionId ?? "",
+      versionId: version.id,
     });
-  }, [taskCount, childPatches, versionId]);
+  }, [taskCount, childPatches, version.id]);
 
   const activeTabs = useMemo(
-    // @ts-expect-error: FIXME. This comment was added by an automated script.
-    () => Object.keys(allTabs).filter((t) => tabIsActive[t] as PatchTab[]),
+    () =>
+      (Object.keys(allTabs) as VersionPageTabs[]).filter((t) => tabIsActive[t]),
     [allTabs, tabIsActive],
   );
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  const isValidTab = tabIsActive[tab];
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  const [selectedTab, setSelectedTab] = useState(activeTabs.indexOf(tab));
-  const previousTab = usePrevious(selectedTab);
+  const [selectedTab, setSelectedTab] = useState(tab || VersionPageTabs.Tasks);
```

Comment:
Refactored so that the selectedTab is no longer an index and actually references a tab name.